materialize-boilerplate/validate.go

```diff
@@ -203,26 +185,87 @@ func (v Validator) validateMatchesExistingBinding(
 					),
 				}
 			}
+		} else if ambiguousFields := v.ambiguousFields(p, boundCollection.Projections); len(ambiguousFields) > 0 {
+			// Projections that would result in ambiguous materialized fields are forbidden if a
+			// different ambiguous projection has already been selected, or optional if none have
+			// yet been selected.
+			if existing != nil && slices.Contains(existing.FieldSelection.AllFields(), p.Field) {
+				// This field has already been selected as the ambiguous field to materialize.
+				c = &pm.Response_Validated_Constraint{
+					Type:   pm.Response_Validated_Constraint_LOCATION_RECOMMENDED,
+					Reason: "This location is part of the current materialization",
+				}
+			} else if existing != nil && v.ambiguousFieldIsSelected(p, existing.FieldSelection.AllFields()) {
+				// A different field has been selected, so this one can't be.
+				c = &pm.Response_Validated_Constraint{
+					Type: pm.Response_Validated_Constraint_FIELD_FORBIDDEN,
+					Reason: fmt.Sprintf(
+						"Flow collection field '%s' is ambiguous with fields already being materialized as '%s' in the destination. Consider using an alternate, unambiguous projection of this field to allow it to be materialized",
+						p.Field,
+						v.is.translateField(p.Field),
+					),
+				}
+			} else if c.Type == pm.Response_Validated_Constraint_LOCATION_RECOMMENDED || c.Type == pm.Response_Validated_Constraint_FIELD_OPTIONAL {
+				// None of these ambiguous fields have been selected yet, so it's still possible to
+				// pick one.
+				c = &pm.Response_Validated_Constraint{
+					Type: pm.Response_Validated_Constraint_FIELD_OPTIONAL,
+					Reason: fmt.Sprintf(
+						"Flow collection field '%s' would be materialized as '%s', which is ambiguous with fields [%s]. Only a single field from this set should be selected. Consider using alternate projections if you want to materialize more than one of these fields",
```

Comment:
Noting this error is the same as the one in the new bindings case. _Maybe_ it could be extracted into a common constant, but ... that confuses Go tooling and there's just one instance of duplication so probably best to leave as-is :shrug:. Maybe just comment both to note that they're effectively the same reason and language should be changed together ?